src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -7,1096 +7,13991 @@
+            (D_808535F0 & 0x40) && (this->actor.bgCheckFlags & 1) && (this->wallHeight >= 39.0f)) {
+
+            this->stateFlags2 |= 1;
+
+            if (CHECK_PAD(sControlInput->cur, A_BUTTON)) {
+
+                if ((this->actor.wallPolySource != 50) &&
+                    ((wallPolyActor = DynaPolyInfo_GetActor(&globalCtx->colCtx, this->actor.wallPolySource)) != NULL)) {
+
+                    if (wallPolyActor->actor.id == ACTOR_BG_HEAVY_BLOCK) {
+                        if (Player_GetStrength() < PLAYER_STR_GOLD_G) {
+                            return 0;
+                        }
+
+                        func_80836898(globalCtx, this, func_8083A0F4);
+                        this->stateFlags1 |= 0x800;
+                        this->interactRangeActor = &wallPolyActor->actor;
+                        this->getItemId = GI_NONE;
+                        this->currentYaw = this->actor.wallPolyRot + 0x8000;
+                        func_80832224(this);
+
+                        return 1;
+                    }
+
+                    this->unk_3C4 = &wallPolyActor->actor;
+                } else {
+                    this->unk_3C4 = NULL;
+                }
+
+                func_8083F72C(this, &D_040030F8, globalCtx);
+
+                return 1;
+            }
+        }
+    }
+
+    return 0;
+}
+
+s32 func_8083F9D0(GlobalContext* globalCtx, Player* this) {
+    if ((this->actor.bgCheckFlags & 0x200) && ((this->stateFlags2 & 0x10) || CHECK_PAD(sControlInput->cur, A_BUTTON))) {
+        DynaPolyActor* wallPolyActor = NULL;
+
+        if (this->actor.wallPolySource != 50) {
+            wallPolyActor = DynaPolyInfo_GetActor(&globalCtx->colCtx, this->actor.wallPolySource);
+        }
+
+        if (&wallPolyActor->actor == this->unk_3C4) {
+            if (this->stateFlags2 & 0x10) {
+                return 1;
+            } else {
+                return 0;
+            }
+        }
+    }
+
+    func_80839FFC(this, globalCtx);
+    func_80832264(globalCtx, this, &D_04003100);
+    this->stateFlags2 &= ~0x10;
+    return 1;
+}
+
+void func_8083FAB8(Player* this, GlobalContext* globalCtx) {
+    func_80835C58(globalCtx, this, func_8084B898, 0);
+    this->stateFlags2 |= 0x10;
+    func_80832264(globalCtx, this, &D_040030F0);
+}
+
+void func_8083FB14(Player* this, GlobalContext* globalCtx) {
+    func_80835C58(globalCtx, this, func_8084B9E4, 0);
+    this->stateFlags2 |= 0x10;
+    func_80832264(globalCtx, this, D_80853C5C[this->modelAnimType]);
+}
+
+void func_8083FB7C(Player* this, GlobalContext* globalCtx) {
+    this->stateFlags1 &= ~0x8200000;
+    func_80837B9C(this, globalCtx);
+    this->linearVelocity = -0.4f;
+}
+
+s32 func_8083FBC0(Player* this, GlobalContext* globalCtx) {
+    if (!CHECK_PAD(sControlInput->press, A_BUTTON) && (this->actor.bgCheckFlags & 0x200) &&
+        ((D_808535F0 & 8) || (D_808535F0 & 2) ||
+         func_80041E4C
```

Comment:
you can go with shockTimer on this or something if you want. it spawns the shock effect and plays the shock sound, but doesnt make link do the animation, thats probably handled elsewhere.

unk_892 right next to it gets the current timer value added to it, and then only spawns the effect and makes the sound if the result is greater than 20. psuedo randomness i guess? no idea what to call it based on that